awx_collection/plugins/modules/tower_project.py

```diff
@@ -229,8 +229,8 @@ def main():
         scm_type=dict(choices=['manual', 'git', 'svn', 'insights'], default='manual'),
         scm_url=dict(),
         local_path=dict(),
-        scm_branch=dict(default=''),
-        scm_refspec=dict(default=''),
+        scm_branch=dict(),
+        scm_refspec=dict(),
```

Comment:
Also make sure there are tests for these as well. Lines 322/323 are hard coding these into the project_files unconditionally.

Reply:
Checked this one, and we normally don't reference them, so looks like tests passed. https://github.com/ansible/awx/blob/4e4811870418b17ec173bd7651b675003abb572a/awx_collection/tests/integration/targets/tower_project/tasks/main.yml#L78